src/index.js

```diff
@@ -25,69 +31,81 @@ require('babel-core/register')({
     presets: ['react']
 });
 
-const idyll = (inputPath, opts, cb) => {
-  options = Object.assign({}, {
-    output: 'build',
-    htmlTemplate: '_index.html',
-    componentFolder: 'components',
-    defaultComponents: path.join('components', 'default'),
-    dataFolder: 'data',
-    layout: 'blog',
-    theme: 'idyll',
-    compilerOptions: {
-      spellcheck: true
-    },
-    build: true
-  }, opts || {});
-
-  const inputDirectory = path.dirname(inputPath);
-  // Expose an "idyll" field on package.json
-  const inputPackage = require(path.join(inputDirectory, 'package.json'));
-  const inputConfig = Object.assign({}, {
-    components: {}
-  }, inputPackage.idyll || {});
-  Object.keys(inputConfig.components).forEach(key => {
-    inputConfig.components[changeCase.paramCase(key)] = inputConfig.components[key];
-  });
-
-  const IDL_FILE = inputPath;
-  const TMP_PATH = path.resolve(path.join(inputDirectory, '.idyll'));
-  const BUILD_PATH = path.resolve(options.output);
-
-  if (!fs.existsSync(TMP_PATH)){
-    fs.mkdirSync(TMP_PATH);
-  }
-
-  if (!fs.existsSync(BUILD_PATH)){
-    fs.mkdirSync(BUILD_PATH);
-  }
+const getPathFactory = (basedir) => (p) => {
+  if (!p) return undefined;
+  if (isAbsolute(p)) return p;
+  return join(basedir, p);
+}
+
+const idyll = (options = {}, cb) => {
+  const inputDir = options.inputFile
+    ? isAbsolute(options.inputFile)
+      ? dirname(options.inputFile)
+      : dirname(join(process.cwd(), options.inputFile))
+    : process.cwd();
+  const getPath = getPathFactory(inputDir);
+  const inputPkg = require(getPath('package.json'));
+  const inputCfg = (inputPkg.idyll || {components: {}});
+  for (let key in inputCfg.components) {
```

Comment:
Small style thing - can we keep things like `inputCfg` spelled out like `inputConfig`, this makes the code more friendly and readable.